src/web/utils/utils.ts

```diff
@@ -0,0 +1,208 @@
+import type { NumberProp, TransformProps } from '../../lib/extract/types';
+import {
+  transformsArrayToProps,
+  TransformsStyleArray,
+} from '../../lib/extract/extractTransform';
+import type { BaseProps, CreateComponentProps } from '../../types';
+import { resolve } from '../../lib/resolve';
+import { resolveAssetUri } from '../../lib/resolveAssetUri';
+
+export function parseTransformProp(
+  transform: TransformProps['transform'],
+  props?: BaseProps
+) {
+  const transformArray: string[] = [];
+
+  props && transformArray.push(...stringifyTransformProps(props));
+
+  if (Array.isArray(transform)) {
+    if (typeof transform[0] === 'number') {
+      transformArray.push(`matrix(${transform.join(' ')})`);
+    } else {
+      const stringifiedProps = transformsArrayToProps(
+        transform as TransformsStyleArray
+      );
+      transformArray.push(...stringifyTransformProps(stringifiedProps));
+    }
+  } else if (typeof transform === 'string') {
+    transformArray.push(transform);
+  }
+
+  return transformArray.length ? transformArray.join(' ') : undefined;
+}
+
+export function hasTouchableProperty(props: BaseProps): boolean {
+  return !!(
+    props.onPress ||
+    props.onPressIn ||
+    props.onPressOut ||
+    props.onLongPress
+  );
+}
+
+export function hasResponderEvents(props: BaseProps): boolean {
+  return !!(
+    props.onResponderGrant ||
+    props.onResponderMove ||
+    props.onResponderRelease ||
+    props.onResponderTerminate ||
+    props.onResponderTerminationRequest ||
+    props.onMoveShouldSetResponder ||
+    props.onMoveShouldSetResponderCapture ||
+    props.onResponderEnd ||
+    props.onResponderReject ||
+    props.onResponderStart ||
+    props.onScrollShouldSetResponder ||
+    props.onScrollShouldSetResponderCapture ||
+    props.onSelectionChangeShouldSetResponder ||
+    props.onSelectionChangeShouldSetResponderCapture ||
+    props.onStartShouldSetResponder ||
+    props.onStartShouldSetResponderCapture
+  );
```

Comment:
Are those props easily obtainable, such that we can iterate over them? If not, wouldn't it be better to create array that stores these props and then check it in loop?

Reply:
I mean something like that?
```ts
export function hasResponderEvents(props: BaseProps): boolean {
  const responderEvents = [
    'onResponderGrant',
    'onResponderMove',
    'onResponderRelease',
    'onResponderTerminate',
    'onResponderTerminationRequest',
    'onMoveShouldSetResponder',
    'onMoveShouldSetResponderCapture',
    'onResponderEnd',
    'onResponderReject',
    'onResponderStart',
    'onScrollShouldSetResponder',
    'onScrollShouldSetResponderCapture',
    'onSelectionChangeShouldSetResponder',
    'onSelectionChangeShouldSetResponderCapture',
    'onStartShouldSetResponder',
    'onStartShouldSetResponderCapture',
  ];

  return responderEvents.some((event) => !!props[event as keyof BaseProps]);
}
```

Reply:
Looks ok, but make sure to test that!